### PR TITLE
MULE-18286: Error mappings not honored while running application on L…

### PR DESCRIPTION
**Pull request by gsfernandes**
…azyInit

### CODE REVIEW

**Comment by gsfernandes** [comment]
@afelisatti @lucianoRM I had to made changes in order to fix tests. Nice to see that we have plenty of tests for errors.

**Comment by lucianoRM** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java [hunk 1]
```diff
@@ -466,9 +466,9 @@ private void registerErrorMappings(SpringComponentModel componentModel, Set<Stri
         ComponentIdentifier source = parameters.containsKey(SOURCE_TYPE)
             ? buildFromStringRepresentation(parameters.get(SOURCE_TYPE)) : ANY;
 
-        muleContext.getErrorTypeRepository()
-            .lookupErrorType(source)
-            .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find error '%s'.", source)));
+        if (!muleContext.getErrorTypeRepository().lookupErrorType(source).isPresent()) {
```
Comment (lucianoRM) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java:469]: Why is this? I think that the functional approach is better.

Comment (gsfernandes) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java:469]: https://sonarqube.kbuild.msap.io/coding_rules?open=squid%3AS2201&rule_key=squid%3AS2201

Comment (gsfernandes) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java:469]: 'Return values from functions without side effects should not be ignored'
When the call to a function doesn't have any side effects, what is the point of making the call if the results are ignored? In such case, either the function call is useless and should be dropped or the source code doesn't behave as expected.

Comment (gsfernandes) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java:469]: There is another way to still get the functional approach but makes code more unreadable.

Comment (lucianoRM) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java:469]: ok, make sense.

**Comment by gsfernandes** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Pull request merged**